lms/djangoapps/teams/tests/test_views.py

```diff
@@ -802,6 +818,45 @@ def test_member_trying_to_get_private_topic(self):
         self.assertEqual('private_topic_1_id', result['results'][0]['topic_id'])
         self.assertNotEqual([], result['results'])
 
+    @ddt.unpack
+    @ddt.data(
+        ('student_masters_not_on_team', True),
+        ('student_masters', True),
+        ('student_enrolled', False),
+        ('staff', True),
+    )
+    def test_text_search_organization_protected(self, user, can_see_masters_team):
+        self.reset_search_index()
+        result = self.get_teams_list(
+            data={'text_search': 'master'},
+            user=user,
+        )
+        expected_teams = 1 if can_see_masters_team else 0
+        self.assertEqual(result['count'], expected_teams)
+
+    @ddt.unpack
+    @ddt.data(
+        ('student_on_team_1_private_set_1', True, True, True),
+        ('student_on_team_2_private_set_1', True, True, True),
+        ('student_enrolled', False, False, False),
+        ('staff', True, True, True),
+    )
+    def test_text_search_private_teamset(self, user, can_see_private_1_1, can_see_private_1_2, can_see_private_2_1):
```

Comment:
This is more of a repeat of the above - 'student_enrolled' can't see the teams because he's outside the bubble, but student_on_team_1 should only be able  to see her team and student_on_team_2 should only be able to see hers.

- [ ] add student_masters, he is in the bubble but shouldn't be able to see any private teams